component/prometheus/exporter/blackbox/blackbox_test.go

```diff
@@ -0,0 +1,87 @@
+package blackbox
+
+import (
+	"testing"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/discovery"
+	"github.com/grafana/agent/pkg/river"
+	"github.com/prometheus/common/model"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUnmarshalRiver(t *testing.T) {
+	riverCfg := `
+		config_file = "modules.yml"
+		blackbox_target {
+			name = "target_a"
+			address = "http://example.com"
+			module = "http_2xx"
+		}
+		blackbox_target {
+			name = "target_b"
+			address = "http://grafana.com"
+			module = "http_2xx"
+		}
+`
+	var cfg Config
+	err := river.Unmarshal([]byte(riverCfg), &cfg)
+	require.NoError(t, err)
+	require.Equal(t, "modules.yml", cfg.BlackboxConfigFile)
+	require.Equal(t, 2, len(cfg.BlackboxTargets))
+	require.Contains(t, "target_a", cfg.BlackboxTargets[0].Name)
+	require.Contains(t, "http://example.com", cfg.BlackboxTargets[0].Target)
+	require.Contains(t, "http_2xx", cfg.BlackboxTargets[0].Module)
+}
```

Comment:
Maybe a bit pedantic, but might be good to test the second `blackbox_target` as well
```suggestion
	require.Contains(t, "target_b", cfg.BlackboxTargets[1].Name)
	require.Contains(t, "http://grafana.com", cfg.BlackboxTargets[1].Target)
	require.Contains(t, "http_2xx", cfg.BlackboxTargets[1].Module)
}
```